File: 75 days Challenge Leetcode/Two_Pointers/Day-11/Solution.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int S = s.length(); // Length of the string 's'
        int T = t.length(); // Length of the string 't'
        
        // If 's' is empty, it's automatically a subsequence of any string
        if (s.empty())  
            return true;

        // If 's' is longer than 't', it can't be a subsequence
        if (S > T)
            return false;

        int j = 0; // Pointer to track position in 's'

        // Loop through each character in 't' with 'i' and check against 's'
        for (int i = 0; i < T && j < S; i++) {
            if (s[j] == t[i]) // If characters match
                j++;          // Move to the next character in 's'
        }

        // If 'j' has moved to the end of 's', it means all characters in 's' were found in 't' in order
        return j == S;
    }
};

File: 75 days Challenge Leetcode/Two_Pointers/Day-11/Solution.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        S = len(s)  # Length of string 's'
        T = len(t)  # Length of string 't'

        # If 's' is empty, it's automatically a subsequence of any string
        if not s:
            return True

        # If 's' is longer than 't', it can't be a subsequence
        if S > T:
            return False

        # Pointer 'j' to track position in 's'
        j = 0

        # Loop through each character in 't' with 'i' and check with 's'
        for i in range(T):
            if j < S and s[j] == t[i]:  # If characters match, move 'j' to the next character in 's'
                j += 1

        # If 'j' has moved to the end of 's', it means all characters in 's' were found in 't' in order
        return j == S
